subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/AbstractModuleComponentResolveMetadata.java

```diff
@@ -52,140 +50,68 @@
     private final List<String> statusScheme;
     @Nullable
     private final ModuleSource moduleSource;
-    private final ImmutableMap<String, Configuration> configurationDefinitions;
-    private final VariantMetadataRules variantMetadataRules;
     private final ImmutableList<? extends ComponentVariant> variants;
     private final HashValue contentHash;
     private final ImmutableAttributes attributes;
 
-    // Configurations are built on-demand, but only once.
-    private final Map<String, DefaultConfigurationMetadata> configurations = Maps.newHashMap();
-    private Optional<ImmutableList<? extends ConfigurationMetadata>> graphVariants;
-
-    AbstractModuleComponentResolveMetadata(AbstractMutableModuleComponentResolveMetadata metadata) {
+    public AbstractModuleComponentResolveMetadata(AbstractMutableModuleComponentResolveMetadata metadata) {
         this.componentIdentifier = metadata.getId();
         this.moduleVersionIdentifier = metadata.getModuleVersionId();
         changing = metadata.isChanging();
         missing = metadata.isMissing();
         statusScheme = metadata.getStatusScheme();
         moduleSource = metadata.getSource();
-        configurationDefinitions = metadata.getConfigurationDefinitions();
-        variantMetadataRules = metadata.getVariantMetadataRules();
-        contentHash = metadata.getContentHash();
         attributesFactory = metadata.getAttributesFactory();
+        contentHash = metadata.getContentHash();
```

Comment:
:o: I wonder if we should use a more explicit name here, like `getOriginalContentHash()`, to make it clear that it's not a hash of the _current_, potentially transformed, metadata.

Reply:
Done